src/crepe/system/CollisionSystem.h

```diff
@@ -246,61 +216,36 @@ class CollisionSystem : public System {
 		* \param type The type of collider pair.
 		* \return True if a collision is detected, otherwise false.
 		*/
-	bool get_collision(const CollisionInternal & first_info,
-					   const CollisionInternal & second_info,
-					   CollisionInternalType type) const;
+	bool detect_collision(CollisionInternal & first_info, CollisionInternal & second_info, const CollisionInternalType & type);
 
 	/**
 		* \brief Detects collisions between two BoxColliders.
 		*
-		* \param box1 The first BoxCollider.
-		* \param box2 The second BoxCollider.
-		* \param transform1 Transform of the first object.
-		* \param transform2 Transform of the second object.
-		* \param rigidbody1 Rigidbody of the first object.
-		* \param rigidbody2 Rigidbody of the second object.
-		* \return True if a collision is detected, otherwise false.
+		* \param box1 Information about the first BoxCollider.
+		* \param box2 Information about the second BoxCollider.
+		* \return returns resolution vector if collide otherwise returns {-1,-1}
 		*/
-	bool get_box_box_collision(const BoxCollider & box1, const BoxCollider & box2,
-							   const Transform & transform1, const Transform & transform2,
-							   const Rigidbody & rigidbody1,
-							   const Rigidbody & rigidbody2) const;
+	vec2 get_box_box_detection(const BoxColliderInternal & box1, const BoxColliderInternal & box2) const;
 
 	/**
 	 * \brief Check collision for box on circle collider
 	 *
-	 * \param box1 The BoxCollider
-	 * \param circle2 The CircleCollider
-	 * \param transform1 Transform of the first object.
-	 * \param transform2 Transform of the second object.
-	 * \param rigidbody1 Rigidbody of the first object.
-	 * \param rigidbody2 Rigidbody of the second object.
-	 * \return True if a collision is detected, otherwise false.
+	 * \param box1 Information about the BoxCollider.
+	 * \param circle2 Information about the circleCollider.
+	 * \return returns resolution vector if collide otherwise returns {-1,-1}
```

Comment:
Het valt mij op dit punt pas op dat `{-1, -1}` op meerdere plekken wordt gebruikt als een soort null waarde, maar kan dat niet veel beter `{0, 0}` of `{INFINITY, INFINITY}` zijn? De kans is klein dat hij ooit exact deze waarde zal hebben maar `{-1, -1}` is een zinnige resolutie vector toch?